### ISSUE TITLE
Unregistered class expection for registered class reference

### ISSUE THREAD

**Issue by void-dragon** [DelSkayn/rquickjs#85]
Hi,

if i reference a class in the same "bind" context an exception is generated on the time of definition initialization.

```rust
use rquickjs::{bind, Object};

use crate::context::get_context;

#[bind(object)]
pub mod engine {
    use crate::{
        context::ContextPtr,
        engine::scene::{self, ScenePtr},
    };

    #[quickjs(clonable)]
    pub struct Scene {
        ptr: ScenePtr,
    }

    impl Scene {
        pub fn new(name: String) -> Self {
            Self {
                ptr: scene::Scene::new(name),
            }
        }
    }

    #[quickjs(clonable)]
    pub struct Game {
        pub(crate) ptr: ContextPtr,
    }

    impl Game {
        pub fn register(&self, scene: &Scene) {                              // here is the Reference
            let mut scenes = self.ptr.scenes.blocking_write();
            let name = scene.ptr.read().unwrap().name.clone();
            scenes.insert(name, scene.ptr.clone());
        }
    }
}

pub fn register(globals: &Object) {
    globals.init_def::<Engine>().unwrap();                                  // error happens HERE

    let game = engine::Game { ptr: get_context() };

    globals.set("Game", game).unwrap();
}
```

```
  Finished dev [unoptimized + debuginfo] target(s) in 2.07s
     Running `target/debug/server`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: FromJs { from: "null", to: "prototype", message: Some("Unregistered class") }', src/script/wrapper.rs:40:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

But as far as i can see, this should be possible.
At least it would be highly convenient!
Or did i misunderstand the documentation?

**Comment by DelSkayn** [comment]
Hi!
Thanks for raising the issue.
This indeed seems to be a bug, but not with registering classes.

The library seems to assume that if a type implementing `ClassDef` defines a constructor it will also define a prototype.
`Scene` defines a constructor with `Scene::new` but does not have a prototype because it does not define any functions or specify prototype fields.

This will be fixed.
In the mean time you can fix the panic by defining a any method on `Scene` which takes a `self` argument:

```rust

#[bind(object)]
pub mod engine {
    use crate::{
        context::ContextPtr,
        engine::scene::{self, ScenePtr},
    };

    #[quickjs(clonable)]
    pub struct Scene {
        ptr: ScenePtr,
    }

    impl Scene {
        pub fn new(name: String) -> Self {
            Self {
                ptr: scene::Scene::new(name),
            }
        }
       // Adding this fixes the panic.
       pub fn foo(&self){}
    }
}
```

**Comment by DelSkayn** [comment]
This should be fixed in b0b5535f72bf3d814e0ec57e9dfed169a6260ea3.
